Source/WebService/Blackjack.WebService/ClientApp/src/app/components/play-game-page/play-game.component.ts

```diff
@@ -27,21 +65,160 @@ export class PlayGameComponent implements OnInit {
    * @param chipValue
    */
   public _addToWager(chipValue: number): void {
+    if (chipValue > this.user.balance) {
+      alert('You have insufficient funds for this wager.');
+      return;
+    }
     this.wager += chipValue;
-    this.player.balance -= chipValue;
+    this.user.balance -= chipValue;
   }
 
   /**
    * Returns funds to the player's balance and resets the wager to 0.
    */
   public _clearWager(): void {
-    this.player.balance += this.wager;
+    this.user.balance += this.wager;
     this.wager = 0;
   }
 
-  public async ngOnInit(): Promise<void> {
-    this.player = await this._userApi.getUser('test');
+  /**
+  * Begins the game by allowing players to enter their name and how many decks to play with.
+  * Also determines if either the player, dealer, or both have black jack.
+  */
+  public startGame(): void {
+    let decks: number = parseInt(prompt('How many decks are you playing with?', '1'));
+    while (isNaN(decks) || decks < 1) {
+      decks = parseInt(prompt(`${decks} is not a valid amount, please try again.`, '1'));
+    }
+    this.deck.shuffle(decks);
+    this.player.updateHand(this.deck.deal(2));
+    this.dealer.updateHand(this.deck.deal(2));
+    this.isHidden = false;
+
+    /*Checks for blackjacks. If anyone does, game automatically ends.*/
+    if (this.player.hasBlackJack() && this.dealer.hasBlackJack()) {
```

Comment:
Each of the blocks in this if/else if/else change do the same things.

The only thing that changes is the value of this.nextMove

I would recommend adding a function to do these things that takes in the enum you want to set when it's running